src/backend/distributed/executor/multi_real_time_executor.c

```diff
@@ -136,6 +137,15 @@ MultiRealTimeExecute(Job *job)
 				continue;
 			}
 
+			/* in case the task has intermediate results */
+			if (CheckIfSizeLimitIsExceeded())
+			{
+				failedTaskId = taskExecution->taskId;
+				taskFailed = true;
+				sizeLimitIsExceeded = true;
+				break;
```

Comment:
We should be able to move this out of the while loop and omit the `failedTaskId = ` and `taskFailed` part.